base/math.jl

```diff
@@ -519,26 +525,34 @@ julia> round(exp(im*pi/6), digits=3)
 sin(x::Number)
 
 """
-    cos(x)
+    cos(x::T) where T <: Number -> float(T)
 
 Compute cosine of `x`, where `x` is in radians.
 
+Throws a [`DomainError`](@ref) if `isinf(x)`, returns a `T(NaN)` if `isnan(x)`.
```

Comment:
```suggestion
Throws a [`DomainError`](@ref) if `isinf(x)`, returns `T(NaN)` if `isnan(x)`.
```
There's something about the "a" in "a `T(NaN)`" that sounds very weird to me.

Reply:
The reason I wrote "a" there is because I don't think we guarantee preservation of `NaN`s; since there are different `NaN`s (quiet, signaling, entirely different bitpatterns), I thought it was best to only guarantee "a" `NaN`, instead of the specific `NaN` you get when writing `NaN` in source code (the bitpatterns can differ, which can make a difference for some code, e.g. PropCheck.jl, where I'm doing bitpattern aware hashing of floating point values to explicitly distinguish failure cases due to different `NaN` bitpatterns).

Reply:
I would probably say either `T(NaN)` or a NaN (no backtics)

Reply:
No backticks means not having it highlighted as code in rendered documentation, but I can change that, sure.